edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.Http.Test/RegistryControllerTest.cs

```diff
@@ -298,6 +298,40 @@ public async Task DeleteModuleOnBehalfOf_Success()
             Assert.Empty(actualResponseBytes);
         }
 
+        [Theory]
+        [InlineData("l4", "l3;l4", "l3", "l3;l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l4", "l2;l3;l4", "l2", "l2;l3;l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l4", "l4", "l4", "l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l3", "l4", "l4", "l4;l5", true, HttpStatusCode.Unauthorized)]
```

Comment:
maybe add a case for  [InlineData("l3", "l3;l4", "l4", "l4;l5", true, HttpStatusCode.Unauthorized)]

Reply:
No, that would not be a valid test. 
So the first 2 parameters are what we get from the caller, and we figure out 3 and 4. 
First is the caller Id, 2nd is the auth Chain. 
Target is the 1st element of the auth chain. I just pass it in here to not reuse the product code that gets the target from the chain. So if the chain is "l3;l4" then the target is "l3".

Reply:
ok, then [InlineData("l4", "l5;l4;l3", "l3", "l3;l4;l5", false, HttpStatusCode.Unauthorized)] should be modified to
[InlineData("l4", "l5;l4", "l5", "l5", false, HttpStatusCode.Unauthorized)] ?